File: L-7.py
di={0:'ling',1:'yi',2:'er',3:'san',4:'si',5:'wu',6:'liu',7:'qi',8:'ba',9:'jiu'}
io=input()
p=0
for i in io:
    p+=1
    if i=='-':
        print('fu',end='')
    elif p==1:
        i=int(i)
        print(di[i],end='')
    else:
        i=int(i)
        print("",di[i],end='')
print()

File: L-7.c

#include "stdio.h"
#include "string.h"
int main()
{
    char dic[10][5] = {"ling","yi","er","san","si","wu","liu","qi","ba","jiu"};
    char io[50];
    int len,t;
    scanf("%s",io);
    if (io[0]!='-')
    {
        if (io == "0")
        {
            printf("%s\n", dic[0]);
        }
        else
        {
            len = strlen(io);
            for (size_t i = 0; i < len; i++)
            {
                if (i==0)
                {
                    t = io[i] - 48;
                    printf("%s", dic[t]);
                    continue;
                }
                t = io[i] - 48;
                printf(" %s",dic[t]);
            }
        }
    }
    else
    {
        printf("fu");
        len = strlen(io);
        for (size_t i = 1; i < len; i++)
        {
            t = io[i] - 48;
            printf(" %s", dic[t]);
        }
    }
    return 0;
}